File: backend/src/main/java/com/example/Flicktionary/global/dto/PageDto.java
package com.example.Flicktionary.global.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.data.domain.Page;
import org.springframework.lang.NonNull;

import java.util.List;

@Getter
@AllArgsConstructor
public class PageDto<T> {

    @NonNull
    List<T> items;

    @NonNull
    private int totalPages;

    @NonNull
    private int totalItems;

    @NonNull
    private int curPageNo;

    @NonNull
    private int pageSize;

    public PageDto(Page<T> page) {
        this.items = page.getContent();
        this.totalItems = (int) page.getTotalElements();
        this.totalPages = page.getTotalPages();
        this.curPageNo = page.getNumber() + 1;
        this.pageSize = page.getSize();
    }

}

File: backend/src/main/java/com/example/Flicktionary/global/dto/PageDto.kt
package com.example.Flicktionary.global.dto

import org.springframework.data.domain.Page
import org.springframework.lang.NonNull

data class PageDto<T>(
    @NonNull
    val items: List<T>,

    @NonNull
    val totalPages: Int,

    @NonNull
    val totalItems: Int,

    @NonNull
    val curPageNo: Int,

    @NonNull
    val pageSize: Int,

    @NonNull
    val sortBy: String
) {
    constructor(page: Page<T>) : this(
        items = page.content,
        totalPages = page.totalPages,
        totalItems = page.totalElements.toInt(),
        curPageNo = page.number + 1,
        pageSize = page.size,
        sortBy = page.sort.toString()
    )
}
